pkg/pace.go

```diff
@@ -63,14 +75,39 @@ func (p *pacer) onPhaseElapsed() {
 }
 
 func (p *pacer) waitDuration(t time.Time) time.Duration {
+	now := time.Now()
+
 	// If there are no more phases left, continue with the last phase's rate
 	if p.lastRequestTime.IsZero() {
 		p.lastRequestTime = t
+		p.lastRequestWallTime = now
+		p.phaseStartRequestTime = p.lastRequestTime
+		p.phaseStartWallTime = p.lastRequestWallTime
+	}
+
+	originalDurationFromPhaseStart := t.Sub(p.phaseStartRequestTime)
+	expectedDurationFromPhaseStart := time.Duration(float64(originalDurationFromPhaseStart) / p.phases[0].rate)
+	expectedWallTime := p.phaseStartWallTime.Add(expectedDurationFromPhaseStart)
+
+	if p.ReportInterval > 0 {
+		// reporting is ON
+		for p.nextReport.Before(expectedWallTime) {
+			fmt.Printf("report_time=%s skew_seconds=%f last_request_time=%s rate=%f expected_rps=%d\n",
```

Comment:
@szmglh I think we should print this to STDERR instead of STDOUT

Reply:
Okay, done in 470f8bcfef82a201fd49c67220af64df43b24b45